SW_Flow_lib.c

```diff
@@ -2158,45 +2193,51 @@ void soil_temperature(double airTemp, double pet, double aet, double biomass,
 	if (debug) {
 		swprintf("\nSoil temperature profile values:");
 		for (i = 0; i <= nRgr + 1; i++) {
-			swprintf("\nk %d oldsTempR %f sTempR %f depth %f",
-				i, st->oldsTempR[i], sTempR[i], (i * deltaX)); // *(oldsTempR + i) is equivalent to writing oldsTempR[i]
+			swprintf("\nk %d oldavgLyrTempR %f avgLyrTempR %f depth %f",
+				i, st->oldavgLyrTempR[i], avgLyrTempR[i], (i * deltaX)); // *(oldavgLyrTempR + i) is equivalent to writing oldavgLyrTempR[i]
 		}
 		swprintf("\n");
 	}
 	#endif
 
-	// convert soil temperature of soil temperature profile 'sTempR' to soil profile layers 'sTemp'
-	lyrTemp_to_lyrSoil_temperature(st->tlyrs_by_slyrs, nRgr, st->depthsR, sTempR, nlyrs,
-		st->depths, width, sTemp);
+	// convert soil temperature of soil temperature profile 'avgLyrTempR' to soil profile layers 'avgLyrTemp'
+	lyrTemp_to_lyrSoil_temperature(st->tlyrs_by_slyrs, nRgr, st->depthsR, avgLyrTempR, nlyrs,
+		st->depths, width, avgLyrTemp, st->temperatureRangeR, v->temperatureRange);

```

Comment:
I don't understand why `temperatureRangeR` and `temperatureRange` are variables in structs `ST_RGR_VALUES` and `SW_SOILWAT`, respectively -- they are not used outside the function `soil_temperature()`. Wouldn't it be better that they are local variables?